packages/web-components/src/avatar/avatar.styles.ts

```diff
@@ -0,0 +1,569 @@
+    bottom: 0;
+    right: 0;
+    box-shadow: 0 0 0 ${strokeWidthThin} ${colorNeutralBackground1};
+  }
+
+  :host([size='20']),
+  :host([size='24']) {
+    font-size: ${fontSizeBase100};
+    font-weight: ${fontWeightRegular};
+  }
+
+  :host([size='20']) {
+    width: 20px;
+    height: 20px;
+  }
+
+  :host([size='24']) {
+    width: 24px;
+    height: 24px;
+  }
+
+  :host([size='20']) .default-icon,
+  :host([size='24']) .default-icon,
+  :host([size='20']) ::slotted(svg),
+  :host([size='24']) ::slotted(svg) {
+    width: 16px;
+    height: 16px;
+    font-size: 16px;
+  }
+
+  :host([size='28']) {
+    width: 28px;
+    height: 28px;
+    font-size: ${fontSizeBase200};
+  }
+
+  :host([size='36']) {
+    width: 36px;
+    height: 36px;
+  }
+
+  :host([size='40']) {
+    width: 40px;
+    height: 40px;
+  }
+
+  :host([size='48']),
+  :host([size='56']) {
+    font-size: ${fontSizeBase400};
+  }
+
+  :host([size='48']) {
+    width: 48px;
+    height: 48px;
+  }
+
+  :host([size='48']) .default-icon,
+  :host([size='48']) ::slotted(svg) {
+    width: 24px;
+    height: 24px;
+    font-size: 24px;
+  }
+
+  :host([size='56']) {
+    width: 56px;
+    height: 56px;
+  }
+
+  :host([size='56']) .default-icon,
+  :host([size='56']) ::slotted(svg) {
+    width: 28px;
+    height: 28px;
+    font-size: 28px;
+  }
+
+  :host([size='64']),
+  :host([size='72']),
+  :host([size='96']) {
+    font-size: ${fontSizeBase500};
+  }
+
+  :host([size='64']) .default-icon,
+  :host([size='72']) .default-icon,
+  :host([size='64']) ::slotted(svg),
+  :host([size='72']) ::slotted(svg) {
+    width: 32px;
+    height: 32px;
+    font-size: 32px;
+  }
+
+  :host([size='64']) {
+    width: 64px;
+    height: 64px;
+  }
+
+  :host([size='72']) {
+    width: 72px;
+    height: 72px;
+  }
+
+  :host([size='96']) {
+    width: 96px;
+    height: 96px;
+  }
+
+  :host([size='96']) .default-icon,
+  :host([size='120']) .default-icon,
+  :host([size='128']) .default-icon,
+  :host([size='96']) ::slotted(svg),
+  :host([size='120']) ::slotted(svg),
+  :host([size='128']) ::slotted(svg) {
+    width: 48px;
+    height: 48px;
+    font-size: 48px;
+  }
+
+  :host([size='120']),
+  :host([size='128']) {
+    font-size: ${fontSizeBase600};
+  }
+
+  :host([size='120']) {
+    width: 120px;
+    height: 120px;
+  }
+
+  :host([size='128']) {
+    width: 128px;
+    height: 128px;
+  }
+
+  :host([shape='square']) {
+    border-radius: ${borderRadiusMedium};
+  }
+
+  :host([shape='square'][size='20']),
+  :host([shape='square'][size='24']) {
+    border-radius: ${borderRadiusSmall};
+  }
+
+  :host([shape='square'][size='56']),
+  :host([shape='square'][size='64']),
+  :host([shape='square'][size='72']) {
+    border-radius: ${borderRadiusLarge};
+  }
+  :host([shape='square'][size='96']),
+  :host([shape='square'][size='120']),
+  :host([shape='square'][size='128']) {
+    border-radius: ${borderRadiusXLarge};
+  }
+
+  :host([data-color='brand']) {
```

Comment:
Not sure if there would be a benefit here but would it be worth leveraging `PropertyStyleSheetBehavior` any conditionally add these styles?

Reply:
@eljefe223 For anything net-new we should avoid additional stylesheets and just do a monolithic stylesheet. We don't have evidence that the additional JS and execution benefits us perf wise and so I'd like to keep things very simple unless we need complexity. In that case, we should be smart about the complexity we choose (purposeful and worth any potential trade off).